docs/specs.md

```diff
@@ -87,181 +113,323 @@ can be considered as _a collection of text chunks_. How those chunks are
-* `id`: an arbitrary string that should be unique per document. Its mission is
-  to make it easier later on to map detected PII instances to the chunk they
-  are part of
-* `table`: a 2D array holding the tabular data for this hunk, as a 2-level
-  list of first rows and, inside each row, columns. The contents of each table 
-  position (a table cell) is an arbitrary text string
-* `context`: elements that help to provide additional information to help in
-  the detection and decision process. There can be three types of context, all
-  of them optional:
-   - _global_: it will be a link/transposition of the context in the document
-     header, as described above
-   - _before_: a small section of data from the chunk preceding this one,
-     containing the closest rows to this chunk
-   - _after_: a small section of data from the chunk following this one,
-     containing the closest rows to this chunk
 
-Note that the context element of the chunk is a logical one, and need not be
-present in a static representation of the chunk, or when sent or streamed,
-since it would repeat information unnecessarily. Instead, it would be generated
-on the fly by an appropriate PII processing module, so that processing
-elements down the line have direct access to that context when processing a
-chunk, even when in streaming mode.
+#### Structural iteration
 
+* A document is considered as a sequence of _rows_
+* Each row contains a list of column values, each column value is a string with
+  the contents of the corresponding table cell
+* Iteration then produces a sequence of chunks, one per document **row**.
+* Each chunk contains the three usual fields: `id`, `data` and `context`. 
+* `id` and `context`have the same structure and semantics as for the other
+  document types
+* However, `data` is different: for table documents it contains not a text
+  field, but a _list_ of text fields, one per column in the row. Hence the
+  `data` field can also be iterated upon.
 
 
-#### Sequential source document
+#### Full iteration
 
-This is a simpler document model in which the document is divided into independent chunks, each one with three elements:
+For Table source documents in full iteration mode:
+ * A document is considered as a sequence of _cells_
+ * Cells are traversed in row-major order, top-to-bottom and left-to-right
+ * Traversing produces a linear list of document chunks
+ * A chunk contents has the same elements as in sequence documents: `id`,
+   `data` and `context`
+ * The `data` element contains the contents of a single document cell
+ * The context elements have the usual _global_, _before_ and _after_ fields
+   **plus** one additional field:
+     - _column_: contains the column name and index for the cell in this chunk
+ * Note that here _before_ and _after_ refer to the previous & next cell,
+   which for the first/last cells in a row will be in another row
```

Comment:
In case this is the last cell, maybe the _after_ should be null?

Reply:
The code right now uses the first cell in the next row as the _after_ context for the last column. If we deem this as not useful, we can take it out and remove that _after_ field (and the _before_ field for the first column)